pkg/fission-cli/cmd/client.go

```diff
@@ -101,30 +102,43 @@ func NewClient(opts ClientOptions) (*Client, error) {
 	client := &Client{}
 	var err error
 	var cmdConfig clientcmd.ClientConfig
+	var restConfig *rest.Config
 	if len(opts.Namespace) > 0 {
 		client.Namespace = opts.Namespace
 	} else {
 		cmdConfig, err = GetClientConfig(opts.KubeContext)
 		if err != nil {
-			return nil, err
+			console.Verbose(2, err.Error())
 		}
-		namespace, _, err := cmdConfig.Namespace()
-		if err != nil {
-			return nil, err
+
+		if cmdConfig != nil {
+			namespace, _, err := cmdConfig.Namespace()
+			if err != nil {
+				return nil, err
+			}
+			client.Namespace = namespace
+			console.Verbose(2, "Kubeconfig default namespace %q", namespace)
 		}
-		client.Namespace = namespace
 	}
 
-	console.Verbose(2, "Kubeconfig default namespace %q", client.Namespace)
-
 	if opts.RestConfig != nil {
 		client.RestConfig = opts.RestConfig
-	} else {
+	} else if cmdConfig != nil {
 		restConfig, err := cmdConfig.ClientConfig()
 		if err != nil {
 			return nil, err
 		}
 		client.RestConfig = restConfig
+	} else {
+		console.Verbose(2, "Using in-cluster config")
+		restConfig, err = rest.InClusterConfig()
+		if err != nil {
+			return nil, err
+		}
+
+		client.RestConfig = restConfig
+		client.Namespace = getInClusterConfigamespace()
```

Comment:
This would override namespace passed from CLI. Can you please CLI by passing namespace argument ?

Reply:
This makes fission-cli's default namespace=POD_NAMESPACE.
If namespace is set explicitly then CLI will take it into account.
https://github.com/fission/fission/blob/main/pkg/fission-cli/cmd/cmd.go#L48

Fission-cli is deployed in `fission-function` namespace.
```
/ # ./fission fn list -v 2
couldn't find kubeconfig file. Set the KUBECONFIG environment variable to your kubeconfig's path.
Using in-cluster config
In-cluster config default namespace "fission-function"
NAME ENV EXECUTORTYPE MINSCALE MAXSCALE MINCPU MAXCPU MINMEMORY MAXMEMORY SECRETS CONFIGMAPS NAMESPACE

/ # ./fission fn list -n default -v 2
couldn't find kubeconfig file. Set the KUBECONFIG environment variable to your kubeconfig's path.
Using in-cluster config
In-cluster config default namespace "fission-function"
Namespace for resource default 
NAME     ENV    EXECUTORTYPE MINSCALE MAXSCALE MINCPU MAXCPU MINMEMORY MAXMEMORY SECRETS CONFIGMAPS NAMESPACE
hello-js nodejs poolmgr      0        0        0      0      0         0                            default
```